Splitte import i bydelsfiler

noenandre opened the pull request:
Nå er denne klar for gjennomgang.
Skripter lager nå en fil "bygninger_<kommunenummer>_bydel_<bydelsnavn>.geojson" for vær bydel i byene Oslo, Stavanger, Bergen og Trondheim.
Slik som foreslått i #4

noenandre committed:
Load geometry from overpass.

noenandre committed:
Split cities into subdivision.
Changed geometry functions to support polygons with hole and multipolygons.

noenandre committed:
Renaming in "centroid_area_linear_ring"

NKAmapper commented:
Ser bra ut, men kunne du unngå endringene i building2osm.py ?
Dvs. la building2osm.py lage én fil for hver storby som nå, og så la city_subdivisions.py lese geojson-filen og splitten den til bydelsfiler.

Det gir litt mer fleksibilitet å ha dette helt separat. Kan f.eks. senere utvide til å splitte etter ulike typer polygoner som finnes, f.eks. grunnkretser, postnummer, kirkesogn, skolekretser, valgkretser osv., avhengig av hva som gir mening for en gitt tilfeldig kommune.

noenandre committed:
Resatt building2osm.py

noenandre commented:
Det er selvsagt mulig. Men det blir jo mindre brukervennlig, når city_subdivisions.py må kjøres som en etterbehandling etter building2osm.py. Kan city_subdivisions.py kjøres med subprocess modulen fra building2osm med et kommandoflagg?

NKAmapper commented:
Jeg kan godt legge inn et kommandoflagg og et kall etterpå. Poenget var egentlig å få lagt all logikk inn i city_subdivisions, og å kunne kjøre city_subdivisions som et separat program basert på kommunens geojson bygningsfil.

Her er forresten postnummerområdene, om du vil ta en titt etterpå: https://nedlasting.geonorge.no/geonorge/Basisdata/Postnummeromrader/GML/. Da kan hvilken som helst kommune deles opp. Ser f.eks. at Kristiansand er altfor stor.

Så jeg tenkte at det vanligste use-caset er å kjøre programmet separat der man har behov + at jeg vil kjøre en runde når samtlige kommuner blir generert på nytt, f.eks. en gang i måneden.

noenandre committed:
Flyttet kode over til city_subdivisions.
Rettet feil med centroid_area_linear_ring, etter tips fra turf.js
Rettet tester.

NKAmapper commented:
Skal vi se, hva er argumentene her for å kjøre? Første 5 forsøk feilet.

noenandre commented:
Foreløpig er det <filnavn_med_kommumenummer.geojson> [-s bydeler, postnummer, sogn]

NKAmapper committed:
Noen små justeringer

- La til kommunenavn i filenavn (enklere å søke opp i folderen)
- Erstatte " " med "_" i bydelsnavn
- Tydeligere beskrivelse av input parameter
- La til Drammen (bydelene må mappes

NKAmapper commented:
Flotte saker. Har lastet opp til Jottacloud.

Noen  ting du kan vurdere til neste versjon om du vil:
- Bruke bare kommunenavnet som parameter i stedet for hele filnavnet (evt. filnavn som opsjon). Eksempel i building2osm.py.
- En grovfiltrering med f.eks. bbox på bydelspolygonene vil nok øke farten en del (men ingen krise :).
- Postnummer ser ut til å være mest aktuell/lettest tilgjengelig for oppdeling av andre kommuner.

noenandre commented:
Sjekk mot bbox for bydelspolygonene reduserte kjøretiden på scriptet fra ca 16 minutter til 3,5 minutt for Oslo på min maskin. Så kommer nok en pr. der først.
